src/files/extraction/helpers.ts

```diff
@@ -0,0 +1,202 @@
+import { UnstructuredOpensourceExtraction } from '../entities/extractions/unstructured-opensource-extraction.entity';
+import { UnstructuredAPIExtraction } from '../entities/extractions/unstructured-api-extraction.entity';
+import { nodeQueue, pythonQueue } from '../jobs/extraction.queue';
+
+import { ExtractionBackend } from './constants';
+import { UnstructuredExtractionDocument } from './types';
+
+import { File } from '@/files/entities/file.entity';
+import { EXTRACTION_BACKEND, S3_BUCKET_FILE_STORAGE } from '@/config';
+import { ORM } from '@/database';
+import { QueueName } from '@/jobs/constants';
+
+export function supportsExtraction(
+  mimeType: string,
+  backend: ExtractionBackend = EXTRACTION_BACKEND
+): boolean {
+  switch (backend) {
+    case ExtractionBackend.WDU:
+      return (
+        mimeType.startsWith('text/') ||
+        mimeType.startsWith('image/') ||
+        ['application/json', 'application/pdf'].includes(mimeType)
+      );
+    case ExtractionBackend.UNSTRUCTURED_OPENSOURCE:
+    case ExtractionBackend.UNSTRUCTURED_API: {
+      const extension = mime.getExtension(mimeType);
+      if (!extension) return false;
+      return [
+        'bmp',
+        'csv',
+        'doc',
+        'docx',
+        'eml',
+        'epub',
+        'heic',
+        'html',
+        'jpeg',
+        'png',
+        'md',
+        'msg',
+        'odt',
+        'org',
+        'p7s',
+        'pdf',
+        'png',
+        'ppt',
+        'pptx',
+        'rst',
+        'rtf',
+        'tiff',
+        'txt',
+        'tsv',
+        'xls',
+        'xlsx',
+        'xml'
+      ].includes(extension);
+    }
+  }
+}
+
+export async function scheduleExtraction(
+  file: Loaded<File>,
+  backend: ExtractionBackend = EXTRACTION_BACKEND
+) {
+  switch (backend) {
+    case ExtractionBackend.WDU: {
+      const job = await nodeQueue.add(QueueName.FILES_EXTRACTION_NODE, {
+        fileId: file.id,
+        backend
+      });
+      file.extraction = new WDUExtraction({ jobId: job.id });
+      await ORM.em.flush();
+      break;
+    }
+    case ExtractionBackend.UNSTRUCTURED_OPENSOURCE: {
+      const job = await pythonQueue.add(QueueName.FILES_EXTRACTION_PYTHON, {
+        fileId: file.id,
+        backend
+      });
+      file.extraction = new UnstructuredOpensourceExtraction({ jobId: job.id });
+      await ORM.em.flush();
+      break;
+    }
+    case ExtractionBackend.UNSTRUCTURED_API: {
+      const job = await pythonQueue.add(QueueName.FILES_EXTRACTION_PYTHON, {
+        fileId: file.id,
+        backend
+      });
+      file.extraction = new UnstructuredAPIExtraction({ jobId: job.id });
+      await ORM.em.flush();
+      break;
+    }
+  }
+}
+
+export async function removeExtraction(file: Loaded<File>) {
+  const extraction = file.extraction;
+  if (!extraction) throw new Error('No extraction to remove');
+  switch (extraction.backend) {
+    case ExtractionBackend.WDU:
+      if (extraction.storageId)
```

Comment:
Is this id always different from the main file?

Reply:
Not necessarily, I see your point, will handle it properly during cleanup.